Apps/Src/ReadCarCAN.c

```diff
@@ -252,8 +232,88 @@ void Task_ReadCarCAN(void *p_arg)
             default: 
                 break;
         }
+        
     }
+    
 }
 
+/**
+ * Error handler functions
+ * Passed as callback functions to the main assertTaskError function by assertReadCarCANError
+*/
+
+/**
+ * @brief error handler callback for disabling charging, 
+ * kills contactor and turns off display
+ */ 
+static void handler_ReadCarCAN_chargeDisable(void) {
+
+    // mark regen as disabled and update saturation
+    updateSaturation(-1);
+
+    // Kill contactor using a direct write to avoid blocking calls when the scheduler is locked
+    BSP_GPIO_Write_Pin(CONTACTORS_PORT, ARRAY_CONTACTOR_PIN, false);
 
+    // Check that the contactor was successfully turned off
+    bool ret = (bool)Contactors_Get(ARRAY_CONTACTOR);
+
+    if(ret == false) { // Array contactor was turned off
+
+        // Turn off the array contactor display light
+        UpdateDisplay_SetArray(false);
```

Comment:
This call may result in an OS error, since the scheduler is locked and UpdateDisplay makes calls to Mutex Pend/Post. Instead, let's move this logic somewhere else somehow. I'll leave that for you to figure out.